src/bidiMapper/commandProcessor.ts

```diff
@@ -99,103 +98,101 @@ export class CommandProcessor {
+    commandData: Message.Command
+  ): Promise<Message.CommandResponseResult> {
     switch (commandData.method) {
       case 'session.status':
         return await this._process_session_status(
-          commandData as Session.SessionStatusCommand
+          commandData as Session.StatusCommand
+        );
+      case 'session.subscribe':
+        return await this._process_session_subscribe(
+          commandData as Session.SubscribeCommand
+        );
+      case 'session.unsubscribe':
+        return await this._process_session_unsubscribe(
+          commandData as Session.UnsubscribeCommand
         );
 
       case 'browsingContext.create':
         return await this._contextProcessor.process_browsingContext_create(
-          commandData as BrowsingContext.BrowsingContextCreateCommand
+          commandData as BrowsingContext.CreateCommand
         );
       case 'browsingContext.getTree':
-        return await this._process_browsingContext_getTree(
-          commandData as BrowsingContext.BrowsingContextGetTreeCommand
+        return await this._contextProcessor.process_browsingContext_getTree(
+          commandData as BrowsingContext.GetTreeCommand
         );
       case 'browsingContext.navigate':
         return await this._contextProcessor.process_browsingContext_navigate(
-          commandData as BrowsingContext.BrowsingContextNavigateCommand
+          commandData as BrowsingContext.NavigateCommand
         );
 
       case 'script.callFunction':
         return await this._contextProcessor.process_script_callFunction(
-          commandData as Script.ScriptCallFunctionCommand
+          commandData as Script.CallFunctionCommand
         );
       case 'script.evaluate':
         return await this._contextProcessor.process_script_evaluate(
-          commandData as Script.ScriptEvaluateCommand
+          commandData as Script.EvaluateCommand
         );
 
       case 'PROTO.browsingContext.findElement':
-        return await this._contextProcessor.PROTO_browsingContext_findElement(
-          commandData as BrowsingContext.PROTO.BrowsingContextFindElementCommand
+        return await this._contextProcessor.process_PROTO_browsingContext_findElement(
+          commandData as BrowsingContext.PROTO.FindElementCommand
+        );
+      case 'PROTO.browsingContext.close':
+        return await this._contextProcessor.process_PROTO_browsingContext_close(
+          commandData as BrowsingContext.PROTO.CloseCommand
         );
-      case 'DEBUG.browsingContext.close':
-        return await this._process_DEBUG_Page_close(commandData.params as any);
 
       default:
         throw new Error('unknown command');
     }
   }
 
-  private _onBidiMessage = async (message: BidiCommandMessage) => {
+  private _onBidiMessage = async (message: Message.Command) => {
     try {
       const result = await this._processCommand(message);
 
       const response = {
         id: message.id,
-        result,
+        ...result,
```

Comment:
It's all because of `script. callFunction` and `script.evaluate` can return not `result` but `exceptionDetails`. 

There is an option to throw a custom exception in such a case, but it would be even more ugly.

Reply:
Thanks. I don't think using custom exceptions for protocol-level errors would be any uglier. Command processing functions may throw exceptions at any time like timeouts, implementation bugs, or browser crashes. Since we need to handle these unexpected errors anyway, we might as well use exceptions for "expected" errors too. In any case, this discussion doesn't need to block this PR.